### DISCUSSION TITLE
Dispersive materials missing in Python v0.0.36?

### DISCUSSION THREAD

**Discussion by RFLAH** [thliebig/openEMS-Project#202] [Q&A Usage]
Hello,

first of all I am very excited about this OpenEMS project. I have over 20 years of user experience of commercial 3D EM tools working with RF designs up to mmWs. Now I am trying to build an assistive GUI based on OpenEMS Python API (to use combination of assistive GUI for simulation control, setting up primitives and ports, CSXCAD and Paraview) while trying learn more Python at the same time. 

I have run to few obstacles with Python interface and I am not sure, if this is due to user error or missing features in v0.0.36. One of the most critical for my needs is the missing dispersive materials. I am able to use metal, material and conducting sheet in my code, but it does not accept shape = CSX.AddLorentzMaterial(...). Do I need to import something extra compared to metal, material and conductive sheet? What would be the correct Python syntax for using this property?

I am also following the discussion related to implementing surface roughness to OpenEMS with enthusiasm. Here is a pretty good introduction to the subject [https://www.isola-group.com/wp-content/uploads/PCB-Material-Selection-for-High-speed-Digital-Designs-1.pdf]. In my opinion it is not practical trying to achieve absolute accuracy with the surface roughness model at least in the beginning, because there will always be a difference between simulation and measurement due to variance between PCB laminate lots, laminate weave type, resin content, thickness tolerances etc... It would be good to be able to hit the ballbark with reasonable accuracy still and to be able to adjust and fit this value based on the measurement data. This comes more and more important when going up in frequencies to mmW in RF/high speed HDI PCB designs.

Thank you in advance.

**Comment by thliebig** [comment]
I would need to check but it is more than possible that not all interfaces are yet ported to python. The dispersive materials might very well be among those...

**Reply by RFLAH** [reply]
Thank you Thorsten for your swift reply. It is good to know, so I am not putting too much effort on this feature yet. I tend to easily doubt my Python skills when something is not working as expected, so I am relieved. Is there by the way some priority or preference on which interface the new features are implemented first? I would love to continue using the Python interface due to many other benefits of the Python environment.

**Reply by thliebig** [reply]
It has been a while that I added more python interface functions. Feel free to give it a shot and create a pull request for me to review :smirk:

**Reply by RFLAH** [reply]
I was afraid of something like this :) I am also new in Github and I was kind of hoping to stay on testing and commenting side, but I try to look into this and see, if this is something I could do or better leave for pros. No promises...
